diff --git a/core/database_api.php b/core/database_api.php
index daf1f33..918da3c 100644
--- a/core/database_api.php
+++ b/core/database_api.php
@@ -152,7 +152,9 @@ function db_connect( $p_dsn, $p_hostname = null, $p_username = null, $p_password
 		if( db_is_mysql() ) {
 			# @todo Is there a way to translate any charset name to MySQL format? e.g. remote the dashes?
 			# @todo Is this needed for other databases?
-			db_query( 'SET NAMES UTF8' );
+			db_query( 'SET collation_connection = utf8mb4_unicode_ci' );
+			db_query( 'SET NAMES utf8mb4' );
+			$g_db->utf8mb4 = true;
 		}
 	} else {
 		db_error();
@@ -1267,7 +1269,9 @@ function db_oracle_adapt_query_syntax( $p_query, array &$p_arr_parms = null ) {
  * @return string
  */
 function db_mysql_fix_utf8( $p_string ) {
-	if( !db_is_mysql() ) {
+	global $g_db;
+
+	if( !db_is_mysql() || isset($g_db->utf8mb4) ) {
 		return $p_string;
 	}
 	return preg_replace(
